index.js

```diff
@@ -1,69 +1,37 @@
-var couch2pg = require('./libs/couch2pg/index'),
-    xmlforms = require('./libs/xmlforms/index'),
-    postgrator = require('postgrator'),
-    Promise = require('rsvp').Promise;
+var _ = require('underscore'),
+    couch2pg =  require('./libs/couch2pg/index'),
+    xmlforms = require('./libs/xmlforms/index');
 
-// convert minutes into ms
-var sleepMins = process.env.COUCH2PG_SLEEP_MINS * 60000;
-if (isNaN(sleepMins)) {
-  // no interval specified. Default to once per hour.
+var sleepMs = process.env.COUCH2PG_SLEEP_MINS * 60 * 1000;
+if (isNaN(sleepMs)) {
   console.log('Missing time interval. Defaulting to once per hour.');
-  sleepMins = 3600000;
+  sleepMs = 1 * 60 * 60 * 1000;
 }
 
-var startTime = function() {
-  return new Promise(function (resolve) {
-    var starttime = new Date();
-    console.log('\nStarting import at ' + starttime);
-    resolve(starttime);
-  });
-};
-
-var potentiallyMigrateDatabase = function() {
-  return new Promise(function (resolve, reject) {
-    postgrator.setConfig({
-      migrationDirectory: __dirname + '/migrations',
-      driver: 'pg',
-      connectionString: process.env.POSTGRESQL_URL
-    });
-
-    postgrator.migrate('201505271423', function(err, migrations) {
-      if (err) {
-        reject(err);
-      } else {
-        postgrator.endConnection(function() {
-          resolve(migrations);
-        });
-      }
-    });
-  });
-};
-
 var loop = function () {
-  var starttime;
-  startTime()
-    .then(function (time) {
-      starttime = time;
-    })
-    .then(potentiallyMigrateDatabase)
-    .then(function(migrations) {
-      console.log('Sucessfully migrated', migrations);
+  console.log('Starting loop at ' + new Date());
+  return couch2pg.migrate()
+    .then(function() {
+      console.log('Couch2pg Migration checks complete');
     })
-    .then(couch2pg)
+    .then(couch2pg.import)
     .then(function () {
       console.log('Imported successfully at ' + Date());
     })
-    .then(xmlforms)
+    .then(xmlforms.migrate)
+    .then(function() {
+      console.log('xmlforms Migration checks complete');
+    })
+    .then(xmlforms.extract)
     .then(function () {
       console.log('XML forms completed at ' + Date());
     })
     .catch(function(err) {
-      console.error(err);
-    })
-    .finally(function () {
-      console.log('Next run at ' + new Date(starttime.valueOf() + sleepMins));
+      console.error('Something went wrong', err);
     });
 };
 
-loop();
-setInterval(loop, sleepMins);
+loop().then(function() {
+  console.log('Next run at ' + new Date(new Date().getTime() + sleepMs));
+  setInterval(loop, sleepMs);
+});
```

Comment:
Yep it should resolve medic/medic-webapp#2151 :-) The code inside of `then` should only run once all the code in loop is complete.